pkg/controllers/build/application_controller.go

```diff
@@ -61,38 +59,36 @@ func (r *ApplicationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 			// on deleted requests.
 			return ctrl.Result{}, nil
 		}
-		log.Error(err, "unable to fetch application")
+		log.Error(err, "unable to fetch Application")
 		return ctrl.Result{}, err
 	}
+	application := *(originalApplication.DeepCopy())
 
-	application := originalApplication.DeepCopy()
 	application.SetDefaults(ctx)
 	application.Status.InitializeConditions()
 
-	result, err := r.reconcile(ctx, log, application)
+	result, err := r.reconcile(ctx, log, &application)
 
 	// check if status has changed before updating, unless requeued
 	if !result.Requeue && !equality.Semantic.DeepEqual(application.Status, originalApplication.Status) {
 		// update status
-		log.Info("updating application status", "application", application.Name,
```

Comment:
why not have additional logging may be at debug level?

Reply:
We need to address logging. I tend to do better by thinking about a component of the system in a cross cutting fashion. (Address logging for all controllers at once) rather than address is piecemeal with gap in one controller but not another.

There are other similar concerns, like tests that I'd like to defer until we get basic functionality in place and then address comprehensively.